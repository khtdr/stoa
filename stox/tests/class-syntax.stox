#!/usr/bin/env stox

class Breakfast {
  cook(food) {
    print "Fried eggs and " + food + ".";
  }
  serve(who) {
    print "Enjoy, " + who + ".";
  }
}

class Bagel {}
var bagel = Bagel();

bagel.isReady = false;
print bagel.isReady ? "Bagel is ready" : "... still toasting...";
bagel.isReady = true;
print bagel.isReady ? "Bagel is ready" : "... still toasting...";

class Standard {}

var object = Standard();
object.name = "Standard";
object.meal = Breakfast();

var newBagel = Bagel();
object.meal.toast = newBagel;

object.meal.toast.isReady = false;
print newBagel.isReady ? "Bagel is ready" : "... still toasting...";
newBagel.isReady = true;
print object.meal.toast.isReady ? "Bagel is ready" : "... still toasting...";

// std.meal.cook(std.meal.toast);
// std.meal.serve('customer');
