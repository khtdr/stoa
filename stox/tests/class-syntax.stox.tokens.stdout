CLASS[3,1]
IDENTIFIER[3,7]
LEFT_CURL[3,17]
IDENTIFIER[4,3]
LEFT_PAREN[4,7]
RIGHT_PAREN[4,8]
LEFT_CURL[4,10]
PRINT[5,5]
STRING[5,11]
SEMICOLON[5,24]
RIGHT_CURL[6,3]
IDENTIFIER[7,3]
LEFT_PAREN[7,8]
IDENTIFIER[7,9]
RIGHT_PAREN[7,12]
LEFT_CURL[7,14]
PRINT[8,5]
STRING[8,11]
PLUS[8,21]
IDENTIFIER[8,23]
PLUS[8,27]
STRING[8,29]
SEMICOLON[8,32]
RIGHT_CURL[9,3]
RIGHT_CURL[10,1]
CLASS[12,1]
IDENTIFIER[12,7]
LEFT_CURL[12,13]
RIGHT_CURL[12,14]
IDENTIFIER[13,1]
LEFT_PAREN[13,6]
RIGHT_PAREN[13,7]
SEMICOLON[13,8]
